atc/creds/vault/manager_test.go

```diff
@@ -57,4 +67,157 @@ var _ = Describe("VaultManager", func() {
+				err := json.NewEncoder(w).Encode(api.Secret{
+					Data: map[string]interface{}{"value": "foo"},
+				})
+				Expect(err).ToNot(HaveOccurred())
+			}))
+
+			tlsCert, err := tls.X509KeyPair(serverCertBytes, serverKeyBytes)
+			Expect(err).ToNot(HaveOccurred())
+
+			fakeVault.TLS = &tls.Config{
+				Certificates: []tls.Certificate{tlsCert},
+			}
+
+			fakeVault.StartTLS()
+
+			config = map[string]interface{}{
+				"url":                  fakeVault.URL,
+				"path_prefix":          "/path-prefix",
+				"shared_path":          "/shared-path",
+				"namespace":            "some-namespace",
+				"ca_cert":              string(caBytes),
+				"client_cert":          string(clientCertBytes),
+				"client_key":           string(clientKeyBytes),
+				"server_name":          serverName,
+				"insecure_skip_verify": true,
+				"client_token":         "some-client-token",
+				"auth_backend_max_ttl": "5m",
+				"auth_retry_max":       "15m",
+				"auth_retry_initial":   "10s",
+				"auth_backend":         "approle",
+				"auth_params": map[string]string{
+					"role_id":   "some-role-id",
+					"secret_id": "some-secret-id",
+				},
+			}
+
+			manager = vault.VaultManager{}
+		})
+
+		JustBeforeEach(func() {
+			configErr = manager.Config(config)
+		})
+
+		It("configures TLS appropriately", func() {
+			Expect(configErr).ToNot(HaveOccurred())
+
+			err := manager.Init(lagertest.NewTestLogger("test"))
+			Expect(err).ToNot(HaveOccurred())
+
+			secret, err := manager.Client.Read("some/path")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(secret).ToNot(BeNil())
+			Expect(secret.Data).To(Equal(map[string]interface{}{"value": "foo"}))
+		})
+
+		It("configures all attributes appropriately", func() {
+			Expect(configErr).ToNot(HaveOccurred())
+
+			Expect(manager.URL).To(Equal(fakeVault.URL))
+			Expect(manager.PathPrefix).To(Equal("/path-prefix"))
+			Expect(manager.SharedPath).To(Equal("/shared-path"))
+			Expect(manager.Namespace).To(Equal("some-namespace"))
+
+			Expect(manager.TLS.Insecure).To(BeTrue())
+
+			Expect(manager.Auth.ClientToken).To(Equal("some-client-token"))
+			Expect(manager.Auth.BackendMaxTTL).To(Equal(5 * time.Minute))
+			Expect(manager.Auth.RetryMax).To(Equal(15 * time.Minute))
+			Expect(manager.Auth.RetryInitial).To(Equal(10 * time.Second))
+			Expect(manager.Auth.Backend).To(Equal("approle"))
+			Expect(manager.Auth.Params).To(Equal(map[string]string{
+				"role_id":   "some-role-id",
+				"secret_id": "some-secret-id",
+			}))
+		})
+
+		Context("with optional configs omitted", func() {
+			BeforeEach(func() {
+				delete(config, "path_prefix")
+				delete(config, "auth_retry_max")
+				delete(config, "auth_retry_initial")
+			})
+
+			It("has sane defaults", func() {
+				Expect(configErr).ToNot(HaveOccurred())
+
+				Expect(manager.PathPrefix).To(Equal("/concourse"))
+				Expect(manager.Auth.RetryMax).To(Equal(5 * time.Minute))
+				Expect(manager.Auth.RetryInitial).To(Equal(time.Second))
+			})
+		})
```

Comment:
I was going to say this needs to be separately tested since I noticed the default `path_prefix` missing, but you obviously thought of it first.